QAChat/Data_Processing/document_embedder.py

```diff
@@ -0,0 +1,70 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2023 Jesse Tim Palarus
+# SPDX-FileCopyrightText: 2023 Amela Pucic
+
+import os
+from datetime import datetime
+from enum import Enum
+
+from langchain.embeddings import HuggingFaceInstructEmbeddings
+from langchain.vectorstores import SupabaseVectorStore
+from supabase.client import create_client
+from dateutil import parser
+
+
+class DataSource(Enum):
+    SLACK = "slack"
+    CONFLUENCE = "confluence"
+    DRIVE = "drive"
+    DUMMY = "dummy"
+
+
+class DataInformation:
+    def __init__(self, id: str, last_changed: datetime, typ: DataSource, text: str):
+        self.id = id
+        self.last_changed = last_changed
+        self.typ = typ
+        self.text = text
+
+
+class DocumentEmbedder:
+
+    def __init__(self):
+        self.embedder = HuggingFaceInstructEmbeddings(
+            model_name="hkunlp/instructor-xl",
+        )
+
+        supabase_url = os.environ.get("SUPABASE_URL")
+        supabase_key = os.environ.get("SUPABASE_SERVICE_KEY")
```

Comment:
This is also introduced in the deepL branch. But we can do that in the next sprint

Reply:
In which folder did you create tokens.env?

And does it make sense to save the tokens file in git but without the real tokens?

So that we have something like that in github:
SupabaseKey=...
SLACK_TOKEN=...
SLACK_APP_TOKEN=...

Reply:
tokens.env should be in the QAChat folder. 
I think saving a dummy tokens.env in git seems not that useful right now since either 1. you use the uploaded tokens.env file and have to replace the tokens each time you pull or 2. you don't use it and it's consequently useless.
But maybe I'm overlooking something right now

Reply:
okay I added the loading of the dotenv file in last commit